src/crafting.h

```diff
@@ -104,6 +104,84 @@ struct recipe {
 
 };
 
+/**
+*   enum used by comp_selection to indicate where a component should be consumed from.
+*/
+enum usage {
+    use_from_map = 1,
+    use_from_player = 2,
+    use_from_both = 1 | 2,
+    use_from_none = 4,
+    cancel = 5 // FIXME: hacky.
```

Comment:
The compiler doesn't know what you're trying to do, so it can't do anything
fancy about it.  It just treats it like an integer, and the OR operator is
just being a little fancy.